### PR TITLE
feat: 🎸 upgrade polkadot.js to 8.9.1

### PR DESCRIPTION
**Pull request by polymath-eric**
### Description

Upgrade polkadot.js to get access to BTreeSet typegen supoort

### JIRA Link

[DA-368](https://polymath.atlassian.net/browse/DA-368)

### Checklist

- [ ] Updated the Readme.md (if required) ?

### CODE REVIEW

**Review by prashantasdeveloper — APPROVED** [review]
LGTM

File: src/testUtils/mocks/dataSources.ts [hunk 1]
```diff
@@ -1614,6 +1615,24 @@ export const createMockEcdsaSignature = (
...
+ */
+export const createMockBTreeSet = <T extends Codec>(
+  items: BTreeSet<T> | unknown[]
+): MockCodec<BTreeSet<T>> => {
+  if (isCodec<BTreeSet>(items)) {
+    return items as MockCodec<BTreeSet<T>>;
+  }
+  const res = createMockCodec(items, !items) as unknown as BTreeSet;
+  res.has = sinon.stub().returns(false);
+  (res as any).size = items.length;
```
Comment (monitz87) [inline src/testUtils/mocks/dataSources.ts:1629]: Why is this assertion necessary?

File: src/testUtils/mocks/dataSources.ts [hunk 2]
```diff
@@ -1614,6 +1615,24 @@ export const createMockEcdsaSignature = (
...
+  items.forEach(i => {
+    (res.has as sinon.SinonStub).withArgs(i).returns(true);
```
Comment (monitz87) [inline src/testUtils/mocks/dataSources.ts:1631]: If you store the stub in a variable before assigning it to the result you can use that and avoid having to use this type assertion

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/testUtils/mocks/dataSources.ts (hunk 1 above)
Comment (polymath-eric) [reply src/testUtils/mocks/dataSources.ts:1629]: I think size is a special readonly property of a BTreeSet, so I used this to set it anyway. I forget the exact reason but will look again in the morning

**Review by VictorVicente — APPROVED** [review]

**Pull request merged**